core/src/main/java/org/mule/runtime/core/routing/EventGroup.java

```diff
@@ -329,11 +328,11 @@ public Event getMessageCollectionEvent() {
             Event.builder(lastEvent).message(builder.build()).session(getMergedSession(muleEvents)).build();
         return muleEvent;
       } else {
-        return VoidMuleEvent.getInstance();
+        return null;//VoidMuleEvent.getInstance();
```

Comment:
I commented out this to look at it later, but then found all tests passed anyway.  I need to review this.